core-v1/src/main/scala/math/Subtract.scala

```diff
@@ -0,0 +1,56 @@
+package com.rallyhealth.vapors.v1
+
+package math
+
+import java.time.{Duration, Instant, LocalDate, Period}
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("""${L} - ${R} is not supported.
+                     
+If these are non-numeric types, try swapping the order of arguments to ${R} - ${L}.
+
+If you think this operation should be allowed, you can define an implicit Subtract[${L}, ${R}].""")
+trait Subtract[L, R] {
+  type Out
+
+  def subtract(
+    left: L,
+    right: R,
+  ): Out
+}
+
+object Subtract extends SubtractNumericImplicits with SubtractJavaTimeImplicits {
+  type Aux[L, R, O] = Subtract[L, R] { type Out = O }
+
+  def apply[L, R, O](fn: (L, R) => O): Subtract.Aux[L, R, O] = new Subtract[L, R] {
+    override type Out = O
+    override def subtract(
+      left: L,
+      right: R,
+    ): Out = fn(left, right)
+  }
```

Comment:
This is not a common use of the `.apply` method. I should rename this to `.instance` or maybe `.define`